Sources/WebAuthn/Ceremonies/Registration/PublicKeyCredentialCreationOptions.swift

```diff
@@ -55,17 +77,87 @@ public struct PublicKeyCredentialCreationOptions: Encodable, Sendable {
         try container.encode(user, forKey: .user)
         try container.encode(relyingParty, forKey: .relyingParty)
         try container.encode(publicKeyCredentialParameters, forKey: .publicKeyCredentialParameters)
-        try container.encodeIfPresent(timeout?.milliseconds, forKey: .timeout)
+        try container.encodeIfPresent(timeoutInMilliseconds, forKey: .timeoutInMilliseconds)
         try container.encode(attestation, forKey: .attestation)
+        try container.encode(authenticatorSelection, forKey: .authenticatorSelection)
+        try container.encode(hints, forKey: .hints)
+        try container.encode(extensions, forKey: .extensions)
+        try container.encode(excludeCredentials, forKey: .excludeCredentials)
+    }
+    
+    let hints: [Hint]
+    enum Hint: String, Encodable
```

Comment:
Rather than an enum, which can never take new values after the library gets a public release, please use UnreferencedStringEnumeration: https://github.com/swift-server/webauthn-swift/blob/75c32a7730188646dfae7c44adcba30242bdf220/Sources/WebAuthn/Ceremonies/Shared/AuthenticatorAttachment.swift#L21

Reply:
Where can I find out more about enums being problematic after public releases? I would like to learn more about this.
Meanwhile, I'll adjust according to your suggestions.

Reply:
The problem is described in the motivation section here: https://forums.swift.org/t/pitch-non-frozen-enumerations/68373

Reply:
Ok, thank you for sharing, I'll read up on that, never thought of enums being risky 😃

Reply:
Done as part of `5f8974df64c68188297a6841e77df797bb11ed70`